simulator/CMakeLists.txt

```diff
@@ -0,0 +1,116 @@
+#cmake
+cmake_minimum_required(VERSION 2.6)
+project(mipt-mips)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic -Wold-style-cast -Werror")
+
+if(APPLE) 
```

Comment:
That does not work.

What about
```
set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)
```

Reply:
Not very clear whats wrong with that. Why does that not work? APPLE is true on OSX

Reply:
Looks like you forget to update rules for test

```
Scanning dependencies of target Test-infra-instrcache
[ 56%] Building CXX object infra/instrcache/t/CMakeFiles/Test-infra-instrcache.dir/unit_test.cpp.o
error: invalid value 'c++17' in '-std=c++17'
make[2]: *** [infra/instrcache/t/CMakeFiles/Test-infra-instrcache.dir/unit_test.cpp.o] Error 1
make[1]: *** [infra/instrcache/t/CMakeFiles/Test-infra-instrcache.dir/all] Error 2
make: *** [all] Error 2
```

Anyway, I would prefer to use CMake's internals instead of that macro to avoid problems when (sooner or later) Apple updates its XCODE compiler.